binding.c

```diff
@@ -1,8 +1,14 @@
-#include <node_api.h>
+// #include <node_api.h>
+// #include <inttypes.h>
+#include <udx.h>
+#include <bare.h>
+#include <js.h>
 #include <stdlib.h>
 #include <string.h>
-#include <udx.h>
-#include <uv.h>
+
+#ifndef JS_AUTO_LENGTH // TODO: is it missing or intentionally removed? (NAPI_AUTO_LENGTH)
+#define JS_AUTO_LENGTH ((size_t) - 1)
+#endif
```

Comment:
```suggestion
```

Pass a literal `-1` instead when we don't have the string length and the string is `NULL` terminated. When we _do_ have the string length, whether the string is `NULL` terminated or not, _always_ pass it.